pandas/core/frame.py

```diff
@@ -4335,6 +4335,9 @@ def drop_duplicates(self, subset=None, keep='first', inplace=False):
         -------
         deduplicated : DataFrame
         """
+        if self.empty:
+            return self.copy()
```

Comment:
instead of this, can you try to replace the last line `return self[-duplicated]` by `return self.iloc[-duplicated]`, for the reasons I mentioned earlier?

Reply:
Then wouldn't index be lost in the case of `DataFrame(index=['A', 'B', 'C'])? As it will be `self.iloc[[]]`?

Reply:
If we use `self.iloc[~duplicated]`, as you said, we will be able to maintain the columns if we have 4 columns and 0 rows. However, if we have 0 columns and 4 rows, we have the same problem - the outcome is `self.iloc[[]]`, which means that the rows won't be selected, for the same reason with using `self[[]]` for columns, thus ending up with 0 columns and 0 rows, once again.

Reply:
I'm not sure whether it'd make sense for a dataframe with rows but no columns to drop the empty rows as duplicate.

But as @jreback if happy with this implementation, just leave it like this. I didn't see his comment earlier.

Reply:
I tried a test, and here's the result:
```
import pandas as pd

a = pd.DataFrame(index=['A', 'B'])
b = a.iloc[[]]
a.shape # (2, 0)
b.shape # (0, 0)
```

Reply:
Yes, that makes sense to me. If we consider that empty rows are equal among them, then `pd.DataFrame(index=['A', 'B']).duplicated()` would return `False, True`, and `.iloc[-duplicated]` would return the first row.

But as I said, I'm happy to keep the original DataFrame as is for this case, as @jreback is happy with it.

Reply:
Oh, if you see the empty rows as equals, then it does make sense. I haven't thought of it that way. Thanks.